src/stateless/slider.vue

```diff
@@ -0,0 +1,540 @@
+                        <div :style="{left: `${position}px`}" class="slider-rail__handle">
+                            <div class="slider-rail__handle-dot"></div>
+                        </div>
+                    </div>
+                </div>
+
+                <div class="slider-helper-text"></div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import Input from './input.vue'
+
+export default {
+    components: {
+        inputElement: Input,
+    },
+    props: {
+        min: { type: Number },
+        max: { type: Number },
+        step: { type: Number },
+        value: { type: Number },
+        theme: { type: String, default: 'dark' },
+        size: { type: String, default: 'normal' },
+        alignment: { type: String, default: 'left' },
+        locale: { type: String, default: 'en-US' },
+        label: { type: String, required: false },
+        limit: { type: Number, required: false },
+        minLabel: { type: String, required: false },
+        maxLabel: { type: String, required: false },
+        unit: { type: String, required: false },
+        disabled: { type: Boolean, default: false },
+    },
+    data () {
+        return {
+            isDomReady: false,
+            isChanged: false,
+            isDragging: false,
+        }
+    },
+    computed: {
+        limitValue () {
+            return this.limit || this.max
+        },
+        minLabelValue () {
+            return this.minLabel || this.min.toLocaleString(this.locale)
+        },
+        maxLabelValue () {
+            return this.maxLabel || this.limitValue.toString().toLocaleString(this.locale)
+        },
+        index () {
+            return (this.value - this.min) / this.step
+        },
+        stepsCount () {
+            return (this.limitValue - this.min) / this.step
+        },
+        stepPercentage () {
+            return 1 / this.stepsCount
+        },
+        position () {
+            return Math.max(0, Math.min(this.index, this.stepsCount)) / this.stepsCount * this.sliderWidth
+        },
+        sliderWidth () {
+            return this.isDomReady ? this.$refs.bar.clientWidth : 0
+        },
+        decimalPlacesCount () {
+            let decimals = this.step.toString().split('.')[1]
+            return decimals ? decimals.length : 0
+        },
+        isWholeNumber () {
+            return this.decimalPlacesCount === 0
+        },
+        lastActiveIndex () {
+            let valueRatio = (this.value - this.min) / (this.limitValue - this.min)
+            return valueRatio * (this.ticksCount - 1)
+        },
+        stateClass () {
+            return {
+                'disabled': this.disabled,
+                'changed': this.isChanged,
+                'dragging': this.isDragging,
+            }
+        },
+        labelsClass () {
```

Comment:
Because we are checking if size is normal I would just inline this in template (e.g. `{ active: lastActiveIndex >= 0 && size === 'normal' }`. If we want to have it as computed because it's easier to test I would rename it to `labelsActive` which would return something like `{ min: true, max: false }` and then in template `{ active: labelsActive.min && size === 'normal' }`

Reply:
I changed it to `labelsActiveClass` but I'm leaving the logic as computed.